functions/zoom-uploader.py

```diff
@@ -67,6 +54,16 @@ class InvalidOpencastSeriesId(Exception):
 def oc_api_request(method, endpoint, **kwargs):
     url = urljoin(OPENCAST_BASE_URL, endpoint)
     logger.info({"url": url, "kwargs": kwargs})
+
+    session = requests.Session()
+    session.auth = HTTPDigestAuth(OPENCAST_API_USER, OPENCAST_API_PASSWORD)
+    session.headers.update(
+        {
+            "X-REQUESTED-AUTH": "Digest",
+            "X-Opencast-Matterhorn-Authentication": "true",
+        }
+    )
+
```

Comment:
This will create a new session for each request. It would be preferable to have one session per run, right? How about creating the session in process_upload or Upload init?

Reply:
ok i redid it so that a Session gets created (and closed) once per invocation